### PR TITLE
[L-3] adds realm integration into the project

### LINKED ISSUE #3: Integrate Realm database
We need local storage for the models

Implement Realm database for that

### PR DESCRIPTION
**Pull request by echirchir**
This change integrates realm database for persistence

Adds a cocoa pods dep and converts all models into Realm objects

Closes #3

### CODE REVIEW

File: Ledger_iOS/.gitignore [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+# Logs and databases #
+######################
+*.log
+*.sql
+*.sqlite
+
+# OS generated files #
+######################
+.DS_Store
+.DS_Store?
```
Comment (aoeldarrat) [inline Ledger_iOS/.gitignore:34]: do we need both of these?

File: Ledger_iOS/Models/PersistenceManager.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    
+    private let defaults = UserDefaults.standard
+    
+    private let recordsKey = "records_key"
+    
+    var records: [Record]? {
+        get { return defaults.array(forKey: recordsKey) as? [Record]}
+        set { defaults.set(newValue, forKey: recordsKey)}
+    }
+}
```
Comment (aoeldarrat) [inline Ledger_iOS/Models/PersistenceManager.swift:20]: Why not just keep the old file and reuse it? No big deal

**Review by aoeldarrat — APPROVED** [review]
lgmt, woulld love to have you walk me through the realm process

File: Ledger_iOS/Models/PersistenceManager.swift (hunk 2 above)
Comment (echirchir) [reply Ledger_iOS/Models/PersistenceManager.swift:20]: I just figured all model-related code should be in the models package?

File: Ledger_iOS/.gitignore (hunk 1 above)
Comment (echirchir) [reply Ledger_iOS/.gitignore:34]: Should definitely exclude these from git.

**Pull request merged**